src/handlers/transfers/index.ts

```diff
@@ -134,66 +135,44 @@ export const handlers: CallHandler[] = [
       });
     },
   },
-  // ERC20 "transferFrom"
+  // Unified ERC20/ERC721 transferFrom with heuristic
   {
     selector: iface.getSighash("transferFrom"),
     handle: (state: StateChange, payments: Payment[], trace: CallTrace) => {
-      // The way to differentiate ERC20 from ERC721 "transferFrom"
-      // is by checking the return value (which is a boolean value
-      // for ERC20 and is missing for ERC721)
-      if (trace.output && trace.output !== "0x") {
-        const args = iface.decodeFunctionData("transferFrom", trace.input);
-        const token = `erc20:${trace.to}`;
+      const args = iface.decodeFunctionData("transferFrom", trace.input);
 
-        adjustBalance(state, {
-          token,
-          address: args.from,
-          adjustment: args.valueOrTokenId.mul(-1),
-        });
-        adjustBalance(state, {
-          token,
-          address: args.to,
-          adjustment: args.valueOrTokenId,
-        });
+      let isErc721 = false;
 
-        payments.push({
-          from: args.from,
-          to: args.to,
-          token,
-          amount: args.valueOrTokenId.toString(),
-        });
-      }
-    },
-  },
-  // ERC721 "transferFrom"
-  {
-    selector: iface.getSighash("transferFrom"),
-    handle: (state: StateChange, payments: Payment[], trace: CallTrace) => {
-      // The way to differentiate ERC20 from ERC721 "transferFrom"
-      // is by checking the return value (which is a boolean value
-      // for ERC20 and is missing for ERC721)
       if (!trace.output || trace.output === "0x") {
```

Comment:
To simplify the changes, can we keep the logic as it is but instead just change the condition:
`(!trace.output || trace.output !== "0x") && !knownNonStandardERC20.includes(trace.to)`